src/main/java/org/jboss/ejb/client/remoting/ChannelAssociation.java

```diff
@@ -161,7 +161,16 @@ EJBReceiverContext getEjbReceiverContext() {
      * @return
      */
     short getNextInvocationId() {
-        return (short) nextInvocationId.getAndIncrement();
+        for (;;) {
+            short nextId = (short) nextInvocationId.getAndIncrement();
+            if (this.waitingMethodInvocations.containsKey(nextId)) {
+                continue;
+            }
+            if (this.waitingFutureResults.containsKey(nextId)) {
+                continue;
+            }
+            return nextId;
+        }
```

Comment:
If you ask me, the comment above by @TomasHofman is exactly what I would want to find in the code itself! :-)